src/Shouldly/Internals/Numerics.cs

```diff
@@ -14,7 +15,7 @@ internal static class Numerics
         /// </summary>
         /// <param name="obj">The object to check</param>
         /// <returns>true if the object is a numeric type</returns>
-        public static bool IsNumericType(Object obj)
+        public static bool IsNumericType([NotNullWhen(true)] object? obj)
```

Comment:
`NotNullWhen` is used to alert the compiler's flow state analysis to this situation:

```cs
nullableThing = ...
if (IsNumericType(nullableThing))
{
    nullableThing.GetHashCode(); // No warning because IsNumericType returned true
}

nullableThing.GetHashCode(); // Compiler warning because `nullableThing` could be null at this point
```